pennylane/resource/error.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Stores classes and logic to define and track algorithmic error in a quantum workflow.
+"""
+from abc import abstractmethod
+
+
+class AlgorithmicError:
+    """Abstract base class representing different types of mathematical errors."""
+
+    def __init__(self, error: float):
+        self.error = error
```

Comment:
Better to have `Args` instead of annotating the argument, `error: float`.

Reply:
I find it more convenient to just have a look at the doctoring to see what the args are.